torch/csrc/jit/tensorexpr/kernel.cpp

```diff
@@ -1713,6 +1713,59 @@ Tensor* TensorExprKernel::computeValue(const torch::jit::Value* v) {
   }
 }
 
+// Return the (lower, upper) loop bounds if they are constants, else nullopt.
+c10::optional<std::pair<int64_t, int64_t>> loopBounds(const For* loop) {
+  auto start = IRSimplifier::simplify(loop->start());
+  auto stop = IRSimplifier::simplify(loop->stop());
+  if (!start->isConstant() || !stop->isConstant()) {
```

Comment:
We should be able to fuse the loops even if the bounds are not constant, as long as they are equal. May be another refinement for later.